server/client.go

```diff
@@ -1092,53 +1097,69 @@ func (c *client) processMsg(msg []byte) {
 		if sub, ok := srv.routeSidQueueSubscriber(c.pa.sid); ok {
 			if sub != nil {
 				mh := c.msgHeader(msgh[:si], sub)
-				c.deliverMsg(sub, mh, msg)
+				if c.deliverMsg(sub, mh, msg) {
+					return
+				}
 			}
-			return
+			isRouteQsub = true
+			// for queue subscription try hard to deliver a message at least once.
+			// Right now we know fo sure that it's a queue subscription and
+			// we didn't make a delivery attempt, because either a subscriber limit
+			// was exceeded or a subscription is already gone.
+			// So, let a code below find yet another matching subscription.
+			// We are at risk that a message might go forth and back
+			// between routes during these attempts, but at the end
+			// it shall either be delivered (at most once) or drop.
+			c.Debugf("Re-sending message of a detached queue sid %s", c.pa.sid)
```

Comment:
This causes concern here, I don't think we are re-sending a message here correct?

Reply:
Again, comment had been previously changed (and this debug statement removed).